simulator/autoware_auto_msgs_adapter/README.md

```diff
@@ -0,0 +1,96 @@
+# autoware_auto_msgs_adapter
+
+This package is used to convert `autoware_msgs` to `autoware_auto_msgs`.
+
+## Purpose
+
+As we transition from `autoware_auto_msgs` to `autoware_msgs`, we wanted to provide flexibility and compatibility for
+users who are still using `autoware_auto_msgs`.
+
+This adapter package allows users to easily convert messages between the two formats.
+
+## Capabilities
+
+The `autoware_auto_msgs_adapter` package provides the following capabilities:
+
+- Conversion of supported `autoware_msgs` messages to `autoware_auto_msgs` messages.
+- Can be extended to support conversion for any message type pairs.
+- Each instance is designed to convert from a single source message type to a single target message type.
+- Multiple instances can be launched to convert multiple message types.
+- Can be launched as a standalone node or as a component.
+
+## Usage
+
+Customize the adapter configuration by replicating and editing the `adapter_control.param.yaml` file located
+in the `autoware_auto_msgs_adapter/config` directory. Example configuration:
+
+```yaml
+/**:
+  ros__parameters:
+    msg_type_target: "autoware_auto_control_msgs::msg::AckermannControlCommand"
+    topic_name_source: "/control/command/control_cmd"
+    topic_name_target: "/control/command/control_cmd_auto"
+```
+
+Set the `msg_type_target` parameter to the desired target message type from `autoware_auto_msgs`.
+
+Make sure that the `msg_type_target` has the correspondence
+in [src/autoware_auto_msgs_adapter_core.cpp](src/autoware_auto_msgs_adapter_core.cpp).
```

Comment:
I think it's a bit hard for users to look directly at the source code. It would be helpful to have a list of message types.
- autoware_auto_control_msgs/msg/AckermannControlCommand

Reply:
My mistake, fixed in: https://github.com/autowarefoundation/autoware.universe/pull/3876/commits/66edca24d3028a10fe4eae95008f80538f14d0f0

Reply:
To address this, I've made following changes in https://github.com/autowarefoundation/autoware.universe/pull/3876/commits/38a4085f932ee7754f9fa9dc803bec18955f7859 :
- Added `.schema.json` file to be maintained. (tested on my own vs code too)
- Added logging at the beginning of the node to output available formats.
- Now storing the correspondences between adapter creator functions and the related string in a map object. (Instead of if else blocks)
- Updated the README.md to reflect all these changes.